Kd/during comp

daonguyenk06 opened the pull request.

daonguyenk06 committed:
Update02 20230215

added some things and simplfied some of the rules.

daonguyenk06 committed:
Finishing touches 20232015

daonguyenk06 committed:
Rename Rough draft to Coding Standards and Procedures

daonguyenk06 committed:
Update Coding Standards and Procedures

daonguyenk06 committed:
First commit, moved overview in separate file

daonguyenk06 committed:
Create Overview

daonguyenk06 committed:
Create README.md

daonguyenk06 committed:
Delete Overview

Everything is in Readme

daonguyenk06 committed:
Update Coding Standards and Procedures

Changed wordings

daonguyenk06 committed:
Update Coding Standards and Procedures

daonguyenk06 committed:
Delete During-Competition Procedures

daonguyenk06 committed:
Added duringComp policies

daonguyenk06 commented:
Added Readme and duringComp procedures in the last few lines.

25kingk commented on Coding Standards and Procedures:
```diff
@@ -0,0 +1,58 @@
+           - Code must work as advertised.
+           - Code must not interfere with any other functionality of other codes.
+        - When all introduced problems are resolved, code may merge into main.
+           - needs the approval of at least 1 lead and 3 members present.
```
Don't like the three members present part of this line. Not only is it unclear whether it includes both you and/or the lead in the three members present, but rarely are there three or four people in close proximity of each other in the programming team that aren't actively working on something.

Perhaps changing it so that you need to get the approval of a lead and three other members would be better, since it allows for more one on one interactions and that most of the people "present" actually know what you intend to do.

25kingk replied:
Don't like the '3 members present' part of this line, and the criterion for starting new projects in general. Will talk about more in review

25kingk reviewed (dismissed):
Looks pretty good. Fairly improved from the original rough draft. The only problem I have is with starting new code. Doesn't look like there is much of anything in place to prevent people from starting whatever projects they want to without prior permission or meeting of criterion, perhaps clear penalties for breaking this and other rules would better encourage following of the criteria.

Another thing I dislike is the mention of "3 members present" on the criteria of starting new projects. I see several flaws with this line. while it's made somewhat clear that the lead doesn't count toward the three members, it's made unclear whether you count and what  the proper definition of "present" is for determining when and how present members give consent. Add the fact it's hard for someone to find a group of three to four people that aren't actively working on something that includes a lead and it becomes extremely difficult and nuanced to start new projects.

What I think would be better is changing it so that one has to get the approval of three different members prior to requesting the approval of a lead to start a project. This way the lead isn't the sole deciding factor on whether you start a project, but also has a fair bit of weight on the matter. It also removes the nuances of 'being present' for a new project, and allows for more 1 on 1 interactions with the three people you wish to get approval from to start your project.

42daha commented on Coding Standards and Procedures:
```diff
@@ -0,0 +1,58 @@
+  6. Provide LINKS to the original source of copied code (if any).
+  7. Include LINKS to external references where they will be most helpful (if any).
+  8. Add comments when fixing bugs.
+  9. Use comments to mark incomplete implementations and/or anything that's need to be done (if any).
```
Should this state that the comment should start with "TODO: "?  That way everyone hopefully marks them the same, and so that VS.Code/Visual Studio/<whatever IDE> can readily find them.

For example:
`// TODO: Add a spin flourish to the end of the autonomous routine.`

42daha commented on Coding Standards and Procedures:
```diff
@@ -0,0 +1,58 @@
+        - Notify the programming group via Slack immediately! One of the following may be done:
+            - Revert code to a previous commit from main.
+            - Fix problem, create pull request, host a quick review, and merge into main.
+        - If there's no time to do any of the following above before the next match:
```
Should probably strike the word "following" from this sentence.  It is a little confusing having "... any of the *following above* ..."

daonguyenk06 committed:
Update Coding Standards and Procedures

daonguyenk06 commented:
@25kingk  I really love your idea, Kyle, I will definitely take this into consideration. Thank you for your insight!

daonguyenk06 commented on Coding Standards and Procedures:
```diff
@@ -0,0 +1,60 @@
+  2. Good comments do not excuse unclear code.
+  3. If you can’t write a clear comment, there may be a problem with the code.
+  4. Comments should dispel confusion, not cause it.
+  5. Explain UNUSUAL or NONSTANDARD code in comments.
```
This was a replacement for "Unidiomatic".

daonguyenk06 committed:
Update Coding Standards and Procedures

Changed # of members present under New Projects.
Added requirement to proceed on new project.

daonguyenk06 committed:
Added an exception for "DuringComp"

daonguyenk06 committed:
Update Coding Standards and Procedures
